app/src/main/java/com/example/inharmony/ChatAdapter.java

```diff
@@ -0,0 +1,204 @@
+package com.example.inharmony;
+
+import android.content.Context;
+import android.media.Image;
+import android.util.Log;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.ImageView;
+import android.widget.TextView;
+
+import androidx.annotation.NonNull;
+import androidx.recyclerview.widget.RecyclerView;
+
+import com.bumptech.glide.Glide;
+import com.example.inharmony.tasks.LoadChatTrackTask;
+import com.parse.ParseException;
+import com.parse.ParseFile;
+import com.parse.ParseUser;
+
+import java.util.List;
+
+import kaaes.spotify.webapi.android.SpotifyApi;
+import kaaes.spotify.webapi.android.SpotifyService;
+
+public class ChatAdapter extends RecyclerView.Adapter<ChatAdapter.MessageViewHolder> {
+    private static final String TAG = "ChatAdapter";
+    private List<Message> mMessages;
+    private Context mContext;
+    private String token;
+    private ParseUser mUser;
+
+    private static final int MESSAGE_OUTGOING = 0;
+    private static final int MESSAGE_INCOMING = 1;
```

Comment:
well done! it is a good idea to assign a human-readable label to values that have meaning in context (e.g., 1 does not refer to count/magnitude, rather it refers to a status).